sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -203,7 +203,7 @@ export function toRheaMessage(data: EventData, partitionKey?: string): RheaMessa
   if (data.properties) {
     msg.application_properties = data.properties;
   }
-  if (partitionKey != undefined) {
+  if (partitionKey !== null && partitionKey !== undefined) {
```

Comment:
Why? (For more clarity?)

We have been following this pattern(!=undefined) at so many places.

Reply:
Yeah, I think I heard @xirzec make an argument for why we should be more explicit. I think parts of it were because it is as clear as you can get, and doesn't require someone less familiar with JS to know that `null == undefined` returns true.

That said I created an `isDefined` helper method that I used almost everywhere, I should use it here as well.

Reply:
Okay!